rllib/agents/alpha_star/league_builder.py

```diff
@@ -0,0 +1,356 @@
+                                    for pid in all_opponents
+                                ]
+                            )
+                            opponent = np.random.choice(all_opponents, p=probs)
+                            logger.debug(
+                                f"Episode {episode.episode_id}: AgentID "
+                                f"{agent_id} played by {opponent} (frozen)"
+                            )
+                            return opponent
+
+                    # Learning main exploiter vs (learning main OR snapshot main).
+                    elif type_ == "ME":
+                        if episode.episode_id % 2 == agent_id:
+                            main_exploiter = np.random.choice(
+                                [
+                                    p
+                                    for p in trainable_policies
+                                    if p.startswith("main_ex")
+                                ]
+                            )
+                            logger.debug(
+                                f"Episode {episode.episode_id}: AgentID "
+                                f"{agent_id} played by {main_exploiter} (training)"
+                            )
+                            return main_exploiter
+                        else:
+                            # n% of the time, play against the learning main.
+                            # Also always play againt learning main if no
+                            # non-learning mains have been created yet.
+                            if num_main_policies == 1 or (
+                                np.random.random() < prob_playing_learning_main
+                            ):
+                                main = "main_0"
+                                training = "training"
+                            # 100-n% of the time, play against a non-learning
+                            # main. Opponent main snapshots should be selected
+                            # based on a win-rate-derived probability.
+                            else:
+                                all_opponents = [
+                                    f"main_{p}"
+                                    for p in list(range(1, num_main_policies))
+                                ]
+                                probs = softmax(
+                                    [
+                                        worker.global_vars["win_rates"][pid]
+                                        for pid in all_opponents
+                                    ]
+                                )
+                                main = np.random.choice(all_opponents, p=probs)
+                                training = "frozen"
+                            logger.debug(
+                                f"Episode {episode.episode_id}: AgentID "
+                                f"{agent_id} played by {main} ({training})"
+                            )
```

Comment:
in addition to this msg, I imagine what will be pretty useful for debugging is to track how many matches are played between different type of agents, and just show it.
will help us get a sense of whether league is in the right shape, etc.